### PR TITLE
[Widgets] Add titles to Legacy Widgets

### LINKED ISSUE #24566: Widgets screen: Can't tell which Legacy Widgets are on my screen without selecting them
When I first open the new Widgets screen, I'm greeted with this:

Screen Shot 2020-08-14 at 15 16 07

Because we're hiding the title from the Legacy Widget block until it's selected, it's really hard to tell what widgets I have in my widget areas.

How about we **always** show the title? 

Screen Shot 2020-08-14 at 15 13 03

### PR DESCRIPTION
**Pull request by kevin940726**
## Description

First step towards #24566 - it displays widget titles by default. The next step here would be to default to the preview depending on whether or not data entry is required.

Also refactored `<LegacyWidgetEdit>` to function component.

## How has this been tested?

1. Go to the new widgets screen.
2. Create a new widget or just use the existing ones.
3. Switch between edit and preview mode and also try to move focus out of it.

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-widgets/src/blocks/legacy-widget/edit/index.js [hunk 1]
```diff
@@ -33,14 +32,9 @@ function LegacyWidgetEdit( {
 } ) {
 	const [ hasEditForm, setHasEditForm ] = useState( true );
 	// If the block is not selected, defaults to show the preview, otherwise show the edit form.
-	const [ isPreview, setIsPreview ] = useState( ! isSelected );
+	const [ isPreview, setIsPreview ] = useState( false );
```
Comment (kevin940726) [inline packages/edit-widgets/src/blocks/legacy-widget/edit/index.js:35]: We should also update the comment above.

File: packages/edit-widgets/src/blocks/legacy-widget/editor.scss [hunk 2]
```diff
@@ -38,3 +38,12 @@
 	border: $border-width solid $gray-900;
 	border-radius: $radius-block-ui;
 }
+
+.wp-block:not(.is-selected) > .wp-block-legacy-widget__edit-handler {
+	border: 1px solid #1e1e1e;
+	border-radius: 2px;
```
Comment (kevin940726) [inline packages/edit-widgets/src/blocks/legacy-widget/editor.scss:44]: Should we use the suggestion below instead?

```suggestion
	border: $border-width solid $gray-900;
	border-radius: $radius-block-ui;
```

and we should also remove the rule in L37 too?

File: packages/edit-widgets/src/blocks/legacy-widget/editor.scss [hunk 3]
```diff
@@ -38,3 +38,12 @@
...
+}
+
+.wp-block-legacy-widget__edit-handler.is-hidden {
+	display: none;
```
Comment (kevin940726) [inline packages/edit-widgets/src/blocks/legacy-widget/editor.scss:48]: Display none might have some accessibility issue, not sure if it matters in this case though. We might want to just use the `hidden` HTML attribute?

Comment (adamziel) [reply packages/edit-widgets/src/blocks/legacy-widget/editor.scss:48]: Ah good spot - let's use `<VisuallyHidden>` component instead

**Review by adamziel — APPROVED** [review]

File: packages/edit-widgets/src/blocks/legacy-widget/edit/handler.js [hunk 4]
```diff
@@ -84,12 +84,11 @@ class LegacyWidgetEditHandler extends Component {
...
 
+		const RootComponent = isVisible ? 'div' : VisuallyHidden;
 		return (
-			<div
-				className={ classNames(
-					'wp-block-legacy-widget__edit-handler',
-					{ 'is-hidden': ! isVisible }
-				) }
+			<RootComponent
+				key="edit-handler"
```
Comment (kevin940726) [inline packages/edit-widgets/src/blocks/legacy-widget/edit/handler.js:90]: Do we need this `key`?

**Review by kevin940726 — COMMENTED** [review]
I believe you forgot to [remove a CSS rule](https://github.com/WordPress/gutenberg/pull/25638#discussion_r494901568) 😆 ?

File: packages/edit-widgets/src/blocks/legacy-widget/edit/handler.js (hunk 4 above)
Comment (adamziel) [reply packages/edit-widgets/src/blocks/legacy-widget/edit/handler.js:90]: We do, otherwise re-rendering would re-mount children and break TinyMCE.

**Comment by afercia** [comment]
Thanks for the progress made here. Noted on https://github.com/WordPress/gutenberg/issues/24566#issuecomment-700184647 that the widget titles should be H3 headings. I guess this can be iterated on follow-up PRs but please always consider to use something better than a `<div>` element, when appropriate 🙂

**Pull request merged**